# Disallow querying of slots during SSR (`lwc/no-querying-slots-during-ssr`)

ðŸ’¼ This rule is enabled in the âœ… `recommended` config.

<!-- end auto-generated rule header -->

TODO: full length description & reasoning for the rule

## Rule Details

Examples of **incorrect** code for this rule:

```js
// TODO
```

Examples of **correct** code for this rule:

```js
// TODO
```

<!--
## When Not To Use It

TODO: when not to use it
-->

<!--
## Further Reading

TODO: further reading
-->
